packages/common/src/store/ui/search-users-modal/slice.ts

```diff
@@ -38,13 +38,12 @@ const slice = createSlice({
     },
     searchUsersSucceeded: (
       state,
-      action: PayloadAction<{ userIds: number[] }>
+      action: PayloadAction<{ userIds: number[]; limit: number }>
     ) => {
-      state.userList.userIds = state.userList.userIds.concat(
-        action.payload.userIds
-      )
+      const { userIds, limit } = action.payload
+      state.userList.userIds = state.userList.userIds.concat(userIds)
       state.userList.status = Status.SUCCESS
-      state.userList.hasMore = action.payload.userIds.length > 0
+      state.userList.hasMore = userIds.length === limit
```

Comment:
Just checking that things are alright if another request happens after this and returns no results, in the case where the search results in exactly `limit` items and there aren't any more to return after that.

If you wanted, you could move the `hasMore` calculation into the saga and let it compare the limit it requested against the results that were returned. Then we don't have to pipe the limit through across two action payloads.

Reply:
Good call but gonna punt on this refactor.

I think we're ok bc if the previous request has exactly `limit` items, the next request will have 0 items and it'll just be a no-op and stop refreshing. We send 1 extra request but I think that's ok for now. Just checked with `limit = 1` and it works.